Detectors/TPC/calibration/SpacePoints/src/TrackInterpolation.cxx

```diff
@@ -136,13 +136,13 @@ bool TrackInterpolation::trackPassesQualityCuts(const o2::dataformats::TrackTPCI
   return true;
 }
 
-bool TrackInterpolation::interpolateTrackITSTOF(const std::pair<o2::dataformats::EvIndex<>, o2::dataformats::MatchInfoTOF>& matchTOF)
+bool TrackInterpolation::interpolateTrackITSTOF(const o2::dataformats::MatchInfoTOF& matchTOF)
 {
   // get TPC cluster residuals to ITS-TOF only tracks
   size_t trkIdx = mTrackData.size() - 1;
   auto propagator = o2::base::Propagator::Instance();
-  const auto& matchITSTPC = mITSTPCTrackVecInput->at(matchTOF.first.getIndex());
-  const auto& clTOF = mTOFClusterVecInput->at(matchTOF.second.getTOFClIndex());
+  const auto& matchITSTPC = mITSTPCTrackVecInput->at(matchTOF.getTrackIndex());
+  const auto& clTOF = mTOFClusterVecInput->at(matchTOF.getTOFClIndex());
```

Comment:
At https://github.com/AliceO2Group/AliceO2/blob/0f1a1fab34aad357c689472e7f2326d59c205b96/Detectors/TPC/calibration/SpacePoints/src/TrackInterpolation.cxx#L164, why do you call ``trkTPC.getClusterReference``? ``trkTPC.getCluster`` called at the next line is enough to fetch the cluster and assign sector, row.
The same is on https://github.com/AliceO2Group/AliceO2/blob/0f1a1fab34aad357c689472e7f2326d59c205b96/Detectors/TPC/calibration/SpacePoints/src/TrackInterpolation.cxx#L301

Reply:
Hi @shahor02 ,
thanks a lot for your comments. I thought that the `at` is better readable and the performance loss negligible, but I have changed that and also removed the unnecessary call to `trkTPC.getClusterReference`.
However, I still get weird segmentation falls when calling `clTPC.getTime()` for the `ClusterNative`, although according to gdb the cluster is defined in a meaningful way.
This I need to fix somehow.
I am also looking into using DPL input now.
Cheers,
Ole

Below is the error message I get when trying to get the time information from a TPC cluster.

Program received signal SIGSEGV, Segmentation fault.
o2::tpc::TrackInterpolation::interpolateTrackITSTOF (this=0x7fffffff28d0, matchTOF=...) at /home/oschmidt/alice/sw/SOURCES/O2/v1.2.0/0/Detectors/TPC/calibration/SpacePoints/src/TrackInterpolation.cxx:173
173	    float clTPCtime = clTPC.getTime();
(gdb) bt
#0  o2::tpc::TrackInterpolation::interpolateTrackITSTOF (this=0x7fffffff28d0, matchTOF=...) at /home/oschmidt/alice/sw/SOURCES/O2/v1.2.0/0/Detectors/TPC/calibration/SpacePoints/src/TrackInterpolation.cxx:173
#1  0x00007fffe2caecf5 in o2::tpc::TrackInterpolation::process (this=0x7fffffff28d0) at /home/oschmidt/alice/sw/SOURCES/O2/v1.2.0/0/Detectors/TPC/calibration/SpacePoints/src/TrackInterpolation.cxx:65
#2  0x00007fffd848a5b1 in run_calib_tpc_sp (path="./", inputTracksITSTPC="o2match_itstpc.root", inputTracksTPC="tpctracks.root", inputClustersTPC="tpc-native-clusters.root", inputTracksITS="o2trac_its.root", inputClustersITS="o2clus_its.root", 
    inputMatchesTOF="o2match_tof.root", inputClustersTOF="tofclusters.root", outputFile="residuals_tpc.root", inputGeom="O2geometry.root", inputGRP="o2sim_grp.root") at /home/oschmidt/alice/macros/run_calib_tpc_sp.C:96
#3  0x00007fff3ed51262 in ?? ()
#4  0xbe6d88d2ffff6928 in ?? ()
#5  0x00007fffffff6900 in ?? ()
#6  0x00007fffbf21102b in ?? ()
#7  0x00007fffffff68b0 in ?? ()
#8  0x00007fffffff6888 in ?? ()
#9  0x4330c9e45dacfe10 in ?? ()
#10 0x000055555dad8fb0 in ?? ()
#11 0x2ad0045dc33fe8b3 in ?? ()
#12 0x41693a9500000033 in ?? ()
#13 0x0000000000000000 in ?? ()
(gdb) p clTPC
$1 = (const o2::tpc::ClusterNative &) @0x55556c4aee84: {static scaleTimePacked = 64, static scalePadPacked = 64, static scaleSigmaTimePacked = 32, static scaleSigmaPadPacked = 32, timeFlagsPacked = 145587, padPacked = 183, sigmaTimePacked = 28 '\034', 
  sigmaPadPacked = 22 '\026', qMax = 17, qTot = 64}

Reply:
Hi @martenole 

The overhead of ``->at(i)`` is small but non-0, as for the safety: we don't expect empty slots, if there are, we better get segm.viol. and fix the problem...

For the crash on the cluster access, the code seems to be correct, perhaps you should run valgrind to check it there is no memory corruption before.